distributed-lock-core/src/main/java/com/github/alturkovic/lock/configuration/DistributedLockConfiguration.java

```diff
@@ -49,12 +50,12 @@
 
   @Bean
   @ConditionalOnMissingBean
-  public LockBeanPostProcessor lockBeanPostProcessor(final KeyGenerator keyGenerator,
-                                                     final ConfigurableBeanFactory configurableBeanFactory,
-                                                     final IntervalConverter intervalConverter,
-                                                     final RetriableLockFactory retriableLockFactory,
-                                                     @Autowired(required = false) final TaskScheduler taskScheduler) {
-    final LockBeanPostProcessor processor = new LockBeanPostProcessor(keyGenerator, configurableBeanFactory::getBean, intervalConverter, retriableLockFactory, taskScheduler);
+  public static LockBeanPostProcessor lockBeanPostProcessor(@Lazy final KeyGenerator keyGenerator,
+                                                            @Lazy final LockTypeResolver lockTypeResolver,
```

Comment:
[`DistributedLockConfiguration` already conditionally defines a bean of type `LockTypeResolver`](https://github.com/alturkovic/distributed-lock/blob/master/distributed-lock-core/src/main/java/com/github/alturkovic/lock/configuration/DistributedLockConfiguration.java#L82)
```java
@Bean
@ConditionalOnMissingBean
public LockTypeResolver lockTypeResolver(final ConfigurableBeanFactory configurableBeanFactory) {
  return configurableBeanFactory::getBean;
}
```

Previously, it would not have been possible to provide a different `LockTypeResolver` implementation. That said this change is not directly related to the main fix so I'm happy to split it out.

Reply:
Whoops, you are correct, I had a bug then :) Thanks!